api/config/v1alpha1/envoygateway_types.go

```diff
@@ -135,6 +156,56 @@ type RateLimitRedisSettings struct {
 	URL string `json:"url"`
 }
 
+// Extension defines the configuration for registering an extension to
+// the Envoy Gateway control plane.
+type Extension struct {
+	// Resources defines the set of K8s resources the extension will handle.
+	Resources []*GroupVersionKind `json:"resources,omitempty"`
+
+	// Service defines the configuration of the extension service that the Envoy
+	// Gateway Control Plane will call through extension hooks.
+	Service *ExtensionService `json:"service"`
+}
+
+// ExtensionService defines the configuration for connecting to a registered extension service.
+type ExtensionService struct {
+	// Host define the extension service hostname.
+	Host string `json:"host"`
+
+	// Port defines the port the extension service is exposed on.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default=80
+	Port int32 `json:"port,omitempty"`
+
+	// TLS defines TLS configuration for communication between Envoy Gateway and
+	// the extension service.
+	//
+	// +optional
+	TLS *ExtensionTLS `json:"tls,omitempty"`
+}
+
+// ExtensionTLS defines the TLS configuration when connecting to an extension service.
+type ExtensionTLS struct {
```

Comment:
can this be defined similar to https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.GatewayTLSConfig

Reply:
`GatewayTLSConfig` can only fetch tls certs from secrets. where `ExtensionTLS` makes it possible to fetch from other sources like files or remote secret stores in the future. This would particularly handy for our more security conscious users who don't want to place certs in K8s secrets.

Also certain fields in `GatewayTLSConfig` kinda don't make sense in this context such as `TLSModeType`.

Reply:
I was alluding to the `certRefs` style , later a `File` kind could be supported, wdyt ?

Reply:
making `ExtensionTLS.Secret` `[]SecretObjectReference` which would be the same as `certificateRefs` I think is fine.

Reply:
I've updated so that `ExtensionTLS` now contains a `certificateRef` field similar to `GatewayTLSConfig`.

For now it would only accept K8s Secret references (which the default anyways) but we can hijack `SecretObjectReference.Kind` to also refer to `File` type when we want to support that. i.e to configure setting the tls config to point to a certificate on the filesystem inside the container (e.g. attached as a volume) it can be configured as 
```
tls:
  certificateRef:
    kind: File
    object: path/to/cert/on/filesystem
```

Reply:
It's also just `SecretObjectReference` and not `[]SecretObjectReference` since I can't think of a reason why you want to refer to multiple certificate references in this context.